pkg/services/ngalert/state/cache.go

```diff
@@ -156,6 +156,27 @@ func (rs *ruleStates) expandRuleLabelsAndAnnotations(ctx context.Context, log lo
 	return expand(alertRule.Labels), expand(alertRule.Annotations)
 }
 
+func (rs *ruleStates) deleteStates(predicate func(s *State) bool) []*State {
```

Comment:
`deleteRuleStates` acquires a mutex but `deleteStates` doesn't. Should we add a comment to mention this? I think its odd that one of them is thread-safe and another is not?

Reply:
this method belongs to ruleStates that does not have mutex. deleteRuleStates acquires the lock and calls this method.

Reply:
Oh apologies, I've misread the code!